pkg/engine/loadtargets.go

```diff
@@ -73,26 +91,94 @@ func getTargets(target kyvernov1.ResourceSpec, ctx *PolicyContext, logger logr.L
 		namespace = ctx.Policy.GetNamespace()
 	}
 
+	apiResource, parentAPIResource, _, err := ctx.Client.Discovery().FindResource(target.APIVersion, target.Kind)
+	if err != nil {
+		return nil, err
+	}
+
 	if namespace != "" && name != "" &&
 		!wildcard.ContainsWildcard(namespace) && !wildcard.ContainsWildcard(name) {
-		obj, err := ctx.Client.GetResource(target.APIVersion, target.Kind, namespace, name)
+		// If the target resource is a subresource
+		var obj *unstructured.Unstructured
+		var parentResourceGVR metav1.GroupVersionResource
+		subresourceName := ""
+		if strings.Contains(apiResource.Name, "/") {
```

Comment:
Please extract re-usable logic to a util function, if possible, to make code easier to read.

Reply:
Replaced a few calls with a function.